app/frontend/controllers/verse_list_controller.js

```diff
@@ -0,0 +1,364 @@
+  var isXrefMarker = event.target.classList.contains('sword-xref-marker');
+  var isTag = event.target.classList.contains('tag');
+
+  if (isReferenceVerse &&
+    ((currentTextType == 'xrefs') || (currentTextType == 'tagged_verses'))
+  ) {
+    if (isXrefMarker) {
+      await app_controller.verse_list_popup.initCurrentXrefs(event.target);
+
+      app_controller.openXrefVerses(this.verse_list_popup.currentReferenceVerseBox,
+                                    this.verse_list_popup.currentPopupTitle,
+                                    this.verse_list_popup.currentXrefs);
+
+    } else if (isTag) {
+
+      app_controller.verse_list_popup.initCurrentTag(event.target);
+
+      app_controller.openTaggedVerses(this.verse_list_popup.currentTagId,
+                                      this.verse_list_popup.currentTagTitle,
+                                      this.verse_list_popup.currentReferenceVerseBox);
+
+    }
+  } else {
+    if (isXrefMarker) {
+      let referenceType = "XREFS";
+
+      if (app_controller.optionsMenu._verseListNewTabOption.isChecked &&
+          !getPlatform().isFullScreen()) { // No tabs available in fullscreen!
+        
+        app_controller.verse_list_popup.currentReferenceType = referenceType;
+        await app_controller.verse_list_popup.initCurrentXrefs(event.target);
+        app_controller.verse_list_popup.openVerseListInNewTab();
+      } else {
+        await app_controller.verse_list_popup.openVerseListPopup(event, referenceType);
+      }
+    } else if (isTag) {
+      let referenceType = "TAGGED_VERSES";
+
+      if (app_controller.optionsMenu._verseListNewTabOption.isChecked &&
+          !getPlatform().isFullScreen()) { // No tabs available in fullscreen!
+        
+        app_controller.verse_list_popup.currentReferenceType = referenceType;
+        app_controller.verse_list_popup.initCurrentTag(event.target);
+        app_controller.verse_list_popup.openVerseListInNewTab();
+      } else {
+        await app_controller.verse_list_popup.openVerseListPopup(event, referenceType);
+      }
+    }
+  }
+}
+
+// Re-init application to state without Bible translations
+function onAllTranslationsRemoved() {
+  resetVerseListView();
+  hideVerseListLoadingIndicator();
+  getCurrentVerseList().append("<div class='help-text'>" + i18n.t("help.help-text-no-translations") + "</div>");
+  $('.book-select-value').text(i18n.t("menu.book"));
+}
+
+module.exports = {
+  init,
+  getCurrentVerseListHeader,
+  getCurrentVerseListFrame,
+  getCurrentVerseList,
+  getCurrentSearchProgressBar,
+  getCurrentSearchCancelButtonContainer,
+  hideSearchProgressBar,
+  getCurrentVerseListLoadingIndicator,
+  showVerseListLoadingIndicator,
+  hideVerseListLoadingIndicator,
+  getBibleBookStatsFromVerseList,
+  getFirstVisibleVerseAnchor,
+  resetVerseListView,
+  getVerseListBookNumber,
+  bindEventsAfterBibleTextLoaded,
+  bindXrefEvents,
+  handleReferenceClick
+};
```

Comment:
Same comment as with `referenceVerseController`. There should be consistency with new/refactored code regarding `module.exports`.